README.md
# academiccommons-theme
Theme files for GWLAI's Academic Commons website.  

At the moment, the theme is currently called `lai` on the understanding that it would apply to all LAI sites as the base of the decoupled/headless theme.  Since at the moment I'm making changes only applicable to Academic Commons (e.g. the logo), I'm going to refer to this repo as academiccommons-theme and deal with LAI-wide theming when we come to the decoupled/headless bridge.

[Release Notes](release-notes.md)


release-notes.md
# Release Notes
Started as of September 11, 2018

2018/09/11
* Styling adjustments to make Peer Coaching Events display well in popups/dialogs (related to Drupal-Modules issue 101 and ticketed in this repo at [24](../../issues/24))
* Fix missing "X" button in dialog by removing boostrap.js (also [24](../../issues/24))
* Make dialog work for logged-out/anonymous users via an including in the libraries.yml file (unticketed)

2018/12/15
* Hide homepage h1 visually but still accessible to screen readers (fixes [35](../../issues/35))
* Adjust styling for new Views output of buckets as wrapped in <a>s (and remove JS formerly meant to make these more linky) (fixes [29](../../issues/29))
* More robust output and CSS for individual fellowship pages (fixes [33](../../issues/33))

2019/01/10
* Update twig templates that redirect traffic to custom pages to use the /tutoring URI instead of /peer-coaching (fixes [41](../../issues/41))

2019/02/07
* Hack to allow (along w/ config and AC modules changes) review session information to display in the same popup currently used for Calendly (fixes [44](../../issues/44))

2019/03/04
* Add styling for the GDPR/cookie-related notice (fixes [47](../../issues/47))

2019/04/30
* Remove references to Peer Coaching (associated JS/CSS added to module instead) (fixes [40](../../issues/40))


lai/php/topheader.php
<!--  inline styling will move to stylesheet -->
<style>

  #header-hours {
  float: right;
  font-family: helvetica, arial;
  font-size: .7em;
  }
  #header-hours .handheld { display: none; }
  @media screen and (max-width:720px) {
    #header-hours .handheld { 
      display: inline-block;
      margin-right: 4px; 
    }
    #header-hours .handheld > * {
      color: #fff; 
      display: block;
    }
    #header-hours .handheld-hours { font-size: 1.4em; margin-top: 4px; }
    #header-hours .handheld-hours-links { font-size: 1.2em; margin-top: 6px; text-align: right; }
    #header-hours .handheld-hours-links a { color: #fff; text-decoration: none; font-style: italic; }
    #header-hours .handheld-hours-links a:hover { text-decoration: underline; }

  }
  #header-hours table {
  border-spacing: 0;
  float: right;
  }
  #header-hours table td {
  padding: .3em .4em;
  margin: 0;
  border-left: 4px solid #005581;
  max-width: 52px;
  white-space: initial;
  }
  #header-hours table tr:first-child {
  background-color: #005581;
  color: /*#ffcB2e*/ #fff /*#ffeebb*/;
  }
  #header-hours table tr:first-child td {
  padding-left: 0;
  font-size: 1.1em;
  }
  #header-hours table tr:nth-child(2) {
  background-color: /*#ffcB2e*/ #fff /*#ffeebb*/;
  color: #005581;
  }
  #header-hours table tr:nth-child(3) {
  color: #fff /*#ffeebb*/;
  }
  #header-hours table tr:nth-child(4) td {
  /*background-color: #005581 !important;*/
  color: #fff /*#ffeebb*/ !important;
  text-align: right;
  padding-right: 0;
  padding-top: .4em;
  font-size: 1.1em;
  padding-bottom: 0;
  }
  #header-hours table tr:nth-child(4) td a {
  color: #fff /*#ffeebb*/;
  text-decoration: none;
  }
  #header-hours table tr:nth-child(4) td a:hover {
  text-decoration: underline;
  }
  #header-hours table tr:nth-child(2) td:first-child, table tr:nth-child(3) td:first-child {
  background-color: #fff /*#ffeebb*/;
  color: #005581;
  padding-left: .6em;
  font-weight: bold;
  }
  #header-hours table tr:first-child td:first-child {
  /*border-bottom-right-radius: 0px;*/
  }
  #header-hours table tr:nth-child(3) td:first-child {
  /*border-top-right-radius: 0px;*/
  }
  #header-hours .container {
  padding:7px 0;
  }

</style>



lai/php/lai-footer.php
<!--<footer id="footer">--> 
	
<!--<section class="pane clear" id="local-footer">-->

	<?php 
	  if ($campus == "gelman" || $campus == "none") {
            $streetAdd = "2130 H Street NW";
            $cityAdd = "Washington, DC 20052";
            $phoneAdd = "202.994.6558";
            $phoneAddLink = "12029946558";
            $emailAdd = "<a href='mailto:gelman@gwu.edu'>gelman@gwu.edu</a>";
	  }
	  elseif ($campus == "eckles") {
            $streetAdd = "Eckles Library<br>2100 Foxhall Road, NW";
            $cityAdd = "Washington, DC 20007";
            $phoneAdd = "202.242.6620";
            $phoneAddLink = "12022426620";
            $emailAdd = "<a href='mailto:eckles@gwu.edu'>eckles@gwu.edu</a>";
	  }
          elseif ($campus == "virginia") {
            $streetAdd = "VSTC Library<br>44983 Knoll Square #179";
            $cityAdd = "Ashburn, VA 20147";
            $phoneAdd = "571.553.8230";
            $phoneAddLink = "17037268230";
            $emailAdd = "<a href='mailto:virginia@gwu.edu'>virginia@gwu.edu</a>";
          }
          elseif ($campus == "utlc") {
            $streetAdd = "2130 H Street NW";
            $cityAdd = "Washington, DC 20052";
            $phoneAdd = "202.994.4026";
            $phoneAddLink = "12029944026";
            $emailAdd = "<a href='mailto:utlc@gwu.edu'>utlc@gwu.edu</a>";
          }
	?>

  <!-- MAXS: The Library/LAI footer -->
    <div class="panel-pane pane-views-panes pane-local-footer-gwtoday-footer-panel-pane">
      <div class="view view-local-footer view-id-local_footer view-display-id-gwtoday_footer_panel_pane view-dom-id-35ea9cc01c7b4ca0fc9260235ad89820">
        <div class="view-content">
          <div class="views-row views-row-1 views-row-odd views-row-first views-row-last">
            <div id="gwtoday-local-footer">
              <div id="gwtoday-footer">
                <div class="promotional-four-col">
                  <div class="footer-logo" id="footer-promo-item-1">
                    <div class="field field-name-field-gwtoday-logo field-type-text-long field-label-hidden">
                      <div class="field-items">
                        <div class="field-item even">
                          <p><a href="https://lai.gwu.edu"><img alt="logo: GW Libraries and Academic Innovation" src="<?php print $front_page . drupal_get_path('theme', $themename); ?>/images/logos/gw_iddol_libraries_2c-90.png"></a></p>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div class="footer-address" id="footer-promo-item-2">
                    <div class="field field-name-field-gwtoday-address field-type-text-long field-label-hidden">
                      <div class="field-items">
                        <div class="field-item even">
                          <div>
                            <span style="line-height: 1.5em;"><?php echo $streetAdd; ?><br></span>
                          </div>
                          <div><?php echo $cityAdd; ?><br>
                            <br>
                            <p class="mobile-show"><span class="phone" style="line-height: 1.5em;"><a href='tel:<?php echo $phoneAddLink; ?>' class='tel'><?php echo $phoneAdd; ?></a></span></p>
                            <p class="mobile-hide" style="display: inline-block;"><span class="phone" style="line-height: 1.5em;"><?php echo $phoneAdd; ?></span></p>
                            <p><span style="line-height: 2.5em;"><?php echo $emailAdd; ?></span></p>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div class="footer-social-links" id="footer-promo-item-3">
                    <div class="field field-name-field-gwtoday-social-links field-type-text-long field-label-hidden">
                      <div class="field-items">
                        <div class="field-item even">
                        <?php if ($campus == "gelman" || $campus == "none"): ?>
                          <p><a href="https://twitter.com/gelmanlibrary" alt="logo: Twitter" target="_blank"><img alt="" src="<?php print $front_page . drupal_get_path('theme', $themename); ?>/images/icons/twitter-logo.png" style="width: 33px; height: 26px;"></a></p>
                          <p><a href="https://twitter.com/gelmanlibrary" target="_blank">Follow us on Twitter</a></p>
                          <p><a href="https://www.facebook.com/gelmanlibrary" alt="logo: Facebook" arget="_blank"><img alt="" src="<?php print $front_page . drupal_get_path('theme', $themename); ?>/images/icons/facebook-logo.png" style="width: 33px;"></a></p>
                          <p><a href="https://www.facebook.com/gelmanlibrary">Like us on Facebook</a></p>
                        <?php elseif ($campus == "eckles"): ?>
                          <p><a href="https://twitter.com/eckleslibrary" alt="logo: Twitter" target="_blank"><img alt="" src="<?php print $front_page . drupal_get_path('theme', $themename); ?>/images/icons/twitter-logo.png" style="width: 33px; height: 26px;"></a></p>
                          <p><a href="https://twitter.com/eckleslibrary" target="_blank">Follow us on Twitter</a></p>
                          <p><a href="https://www.facebook.com/eckleslibrary" alt="logo: Facebook" arget="_blank"><img alt="" src="<?php print $front_page . drupal_get_path('theme', $themename); ?>/images/icons/facebook-logo.png" style="width: 33px;"></a></p>
                          <p><a href="https://www.facebook.com/eckleslibrary">Like us on Facebook</a></p>

                        <?php elseif ($campus == "virginia"): ?>
                          <p><a href="https://www.facebook.com/vstclibrary" alt="logo: Facebook" arget="_blank"><img alt="" src="<?php print $front_page . drupal_get_path('theme', $themename); ?>/images/icons/facebook-logo.png" style="width: 33px;"></a></p>
                          <p><a href="https://www.facebook.com/vstclibrary">Like us on Facebook</a></p>

                        <?php elseif ($campus == "utlc"): ?>
                          <p><a href="https://www.youtube.com/user/GWUTLC" alt="logo: YouTube" arget="_blank"><img alt="" src="<?php print $front_page . drupal_get_path('theme', $themename); ?>/images/icons/youtube-logo.png" style="width: 33px;"></a></p>
                          <p><a href="https://www.youtube.com/user/GWUTLC">UTLC on YouTube</a></p>

                        <?php endif; ?>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div class="footer-quick-links" id="footer-promo-item-4">
                    <div class="field field-name-field-gwtoday-quick-links field-type-text-long field-label-hidden">
                      <div class="field-items">
                        <div class="field-item even">
                          <?php if ($campus == "utlc"): ?>
                          <a href="/utlc/contact">Contact UTLC</a><br/>
                          <a href="/about/facilities/maps">Maps &amp; Directions</a><br/>
                          <a href="/accessibility">Accessibility</a><br/>
                          <a href="https://lai.gwu.edu/staff-info">Staff Information</a><Br/>
                          <a href="/user">Staff Login</a><br/>
                          <?php else: ?>
                          <a href="/hours">Hours</a><br>
                          <a href="/contact">Contact Us</a><br>
                          <a href="/about/feedback">Send Feedback</a><br>
                          <a href="/about/facilities/maps">Maps &amp; Directions</a><br>
                          <a href="http://library.gwu.edu/accessibility">Accessibility</a><br>
                          <a href="http://go.gwu.edu/gwlibraries">Give to GW Libraries</a><br>
                          <a target="_blank" href="https://lai.gwu.edu/staff-info">Staff Information</a><br>
                          <a href="https://<?php print $_SERVER['SERVER_NAME']; ?>/user">Staff Login</a><br>
                          <?php endif; ?>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>


<!-- MAXS: The blue GW-wide footer -->
<footer class="region region-footer" id="footer">
  <div id="footer-wrapper">
    <div class="footer-logof">
      <a href="http://www.gwu.edu/" rel="home" alt="logo: the George Washington University" target="_blank"><img alt="GW logo" height="42" src="<?php print $front_page . drupal_get_path('theme', $themename); ?>/images/logos/gw_txh_2cs_rev.png" typeof="foaf:Image" width="300"></a>
    </div>
    <div class="row-one">
      <ul>
        <li>
          <a href="http://www.campusadvisories.gwu.edu/" target="_blank">Campus Advisories</a>
        </li>
        <li>
          <a href="http://my.gwu.edu/files/policies/EqualEmploymentOpportunityStatement.pdf" target="_blank">Non-Discrimination Policy</a>
        </li>
        <li>
          <a href="http://www.gwu.edu/privacy-policy" target="_blank">Privacy Policy</a>
        </li>
      </ul>
    </div>
    <div class="row-two">
      <ul>
        <li>
          <a href="http://www.gwu.edu/contact-gw" target="_blank">Contact GW</a>
        </li>
        <li>
          <a href="http://www.gwu.edu/terms-use" target="_blank">Terms of Use</a>
        </li>
        <li>
          <a href="http://www.gwu.edu/copyright" target="_blank">Copyright</a>
        </li>
        <li>
          <a href="http://www.gwu.edu/az-index" target="_blank">A-Z Index</a>
        </li>
      </ul>
    </div>
  </div>
</footer>

	<!-- responsive opt-out links -->
	<div>
	  <?php echo $_SESSION['respOptString'] ?>
	</div>


lai/php/lai-header-in-search.php
<div id="internal-search">
	  <form id="search-form">
		<div id="search-dropdown">
		  <div id="current-search-text"><a href="#" aria-haspopup="true" aria-controls="home-search-dropdown-ul" id="current-search-text-a">Search All</a></div>
		  <ul id="home-search-dropdown-ul" aria-labelledby="current-search-text-a">
			<li id="search-all" data-placeholder="<?php echo $bentoPlaceholder; ?>">
			  <a href="#">
		        <span class="search-label">Search All</span>
		        <div class="search-description">Articles and books, plus library databases, research guides and tutorials</div>
			  </a>
		    </li>
		    <li id="search-articlesplus" data-placeholder="Fidel Castro, sustainable energy, gender and identity ...">
			  <a href="#">
		        <span class="search-label">ArticlesPlus</span>
			    <div class="search-description">Journal &amp; newspaper articles, plus books and more</div>
			  </a>
		    </li>
		    <li id="search-catalog" data-placeholder="The Communist Manifesto, calculus, Blade Runner ...">
		      <a href="#">
		        <span class="search-label">Books & More</span>
				<div class="search-description">Books (including e-books), A/V media, and archival resources</div>
			  </a>
			</li>
			<li id="search-journals" data-placeholder="Wall Street Journal, Journal of American History, sociology, ...">
		      <a href="#">
				<span class="search-label">Browse Journals</span>
				<div class="search-description">Online access to journals and other periodicals, by subject area and title</div>
			  </a>
			</li>
			<li id="search-website" data-placeholder="building hours, study rooms, Churchill ...">
		      <a href="#">
				<span class="search-label">Library Website</span>
				<div class="search-description">Library policies, news and events, and research help</div>
			  </a>
		    </li>
		  </ul>
		</div>
	    <input type="text" aria-label="searchbox: enter your search terms here" placeholder="<?php echo $bentoPlaceholder; ?>" />
		<select id="catalog-options">
		  <option>title</option>
		  <option>journal title</option>
		  <option selected="selected">keyword</option>
		  <option>author</option>
		  <option>subject</option>
		  <option>call number</option>
		</select>
		<select id="journals-options">
		  <option>title begins with</option>
		  <option>title (exact)</option>
		  <option selected="selected">title keywords</option>
		  <option>ISSN</option>
		</select>
		<input type="image" src="<?php print $front_page . drupal_get_path('theme', $themename); ?>/images/search.png" alt="Submit search"/>
	  </form>
	</div>

 <script type="text/javascript">

  jQuery("#search-dropdown").on("click", "li", function() {
	  jQuery("#current-search-text-a").text(jQuery(this).find(".search-label").text());
	  jQuery("#search-form input[type=text]").attr("placeholder",jQuery(this).data("placeholder"));
	  jQuery("#home-search-explanation").html(jQuery(this).find(".search-description").html());
	  jQuery("#search-dropdown ul").hide();
	  jQuery("#search-form input[type=text]").focus();
	  jQuery("#catalog-options").hide();
	  jQuery("#journals-options").hide();
	  if (this.id == "search-catalog") {
	    jQuery("#catalog-options").show();
	  }
	  if (this.id == "search-journals") {
	    jQuery("#journals-options").show();
	  }
  });

	jQuery("#search-dropdown").on("click", "a", function(e) {
	  e.preventDefault();
	});

	jQuery("#search-dropdown").on("keydown", function(e) {
	  var ul = jQuery(this).find("ul");
	  if (ul.is(":hidden")) {
	    ul.show();
	  }
	  var searchLabels = jQuery("#internal-search a");
	  var nearestLabel = searchLabels.filter(":focus");
	  var nearestLabelIndex = searchLabels.index(searchLabels.filter(":focus"));
	  if (e.which == 38) { // Up arrow
	    if (nearestLabel.length == 0) {
	      searchLabels.last().focus();
	    } else {
	      searchLabels.eq(nearestLabelIndex - 1).focus();
	    }
	    e.preventDefault(); // page scrolling
	  }
	  if (e.which == 40) { // Down arrow
	    if (nearestLabelIndex == searchLabels.length - 1) {
	      searchLabels.first().focus();
	    } else {
	      searchLabels.eq(nearestLabelIndex + 1).focus();
	    }
	    e.preventDefault(); // page scrolling
	  }
	});

	jQuery("#search-dropdown").on("mouseenter mouseover", function(e) {
	  jQuery(this).find("ul").show();
	});
	jQuery("#search-dropdown").on("mouseleave", function(e) {
	  jQuery(this).find("ul").hide();
	});


  </script>


lai/php/lai-header.php
<?php 
if ($variables['is_front']) {
    $campus = 'none';
  }
  else if (false !== stripos($_SERVER["REQUEST_URI"], 'eckles')) {
    $campus = 'eckles';
  }
  else if (false !== stripos($_SERVER["REQUEST_URI"], 'virginia')) {
    $campus = 'virginia';
  }
  else if (false !== stripos($_SERVER["REQUEST_URI"], 'utlc')) {
          $campus = 'utlc';
  }
  else {
    $campus = 'gelman';
  }
?>

<?php if ( ($_SERVER['SERVER_NAME']=="backup.library.gwu.edu") || ($_SERVER['SERVER_NAME']=="dev.library.gwu.edu") ): ?>
  <div style="background-color:#FFFF00">
    <div class="staging-header-message">
      <span style="color:#FC2847"><?php echo $_SERVER['SERVER_NAME'];?></span>
    </div>
  </div>
<?php endif; ?>


<div id="topheader-container" <?php if($campus == 'eckles' || $campus == 'virginia'): ?>class="is-branch"<?php endif; ?>>

  <div id="topheader-content">

    <?php if ($campus == 'utlc'): ?>
    <div id="gwlogo-topheader" style="width: 460px;">
      <a href="/utlc" aria-label="UTLC home page">
        <img src="<?php print $front_page . drupal_get_path('theme', $themename); ?>/images/logos/utlc-logo-01.png" alt="logo: University Teaching and Learning Center" style="width: 460px;">
        <span class="sr-only">UTLC home page</span>
      </a>
    </div>
    <?php else: ?>
    <div id="gwlogo-topheader">
      <a href="<?php print $front_page; ?>" aria-label="GW Libraries home page">
        <img src="<?php print $front_page . drupal_get_path('theme', $themename); ?>/images/logos/gw_iddol_libraries_1cs_pos.png" alt="logo: The George Washington University Libraries" width="199" height="90">
        <span class="sr-only">GW Libraries home page</span>
      </a>
    </div>
    <?php endif; ?>

    <p id="lai-division">
      A division of <a target="_blank" href="https://lai.gwu.edu">Libraries and Academic Innovation</a>
    </p>

    <?php if ($campus == 'gelman' OR $campus == 'none'): ?>
      <div id='header-hours'>
        <?php print render($page['topheader_right']); ?>
      </div>
    <?php endif; ?> 


<div class="limiter">
  <div id="local-header">   
  </div>
</div>

<!-- Emergency Notice Banner -->

<?php if ($page['emergency_banner']): ?>
  <style type="text/css">
    #topheader-content { height: initial; }
    @media screen and (max-width: 960px) {
      #bannerContainer { margin: 1em auto .5em; }
    }
  </style>
  <div id="bannerContainer">
    <div id="bannerMessage">
      <?php print render($page['emergency_banner']); ?>
    </div>
  </div>
<?php endif; ?>

</div>


<!-- Primary Navigation -->

<?php
include 'libnav.php';
?>
    

<script>document.createElement('main'); //for IE compatibility with semantic elements</script>

<main id="maincontent" class="main-container center-content">
  <div id="hero">
  <?php if(($campus == 'gelman' || $campus == 'none') && drupal_is_front_page()): ?>
    <div id="hero-image">
      <img alt="background image: Gelman library" src="<?php print $front_page . drupal_get_path('theme', $themename); ?>/images/editorial-bg-23.jpg" />
    </div>
  <?php elseif(($campus == 'gelman' || $campus == 'none') && !drupal_is_front_page()): ?>
    <img alt="background image of Gelman Library" src="<?php print $front_page . drupal_get_path('theme', $themename); ?>/images/gelman-10.jpg"/>
  <?php elseif($campus == 'eckles'): ?>
    <div id="hero-image">
      <img alt="background image of Eckles Library" src="<?php print $front_page . drupal_get_path('theme', $themename); ?>/images/eckles-12.jpg"/>
      <div id="branch-name">
        Eckles Library at the Mount Vernon Campus
      </div>
    </div>
  <?php elseif($campus == 'virginia'): ?>
    <div id="hero-image">
      <img alt="background image of Virginia Campus Library" src="<?php print $front_page . drupal_get_path('theme', $themename); ?>/images/virginia-04.jpg"/>
      <div id="branch-name">
        Virginia Science and Technology Campus Library
      </div>
    </div>
  <?php elseif($campus == 'utlc' && request_path() == 'utlc'): ?>
    <div id="panels-ipe-paneid-3" class="panels-ipe-portlet-wrapper panels-ipe-portlet-marker">
      <div class="panels-ipe-portlet-content">
        <div class="panel-pane pane-views-panes pane-promotion-slider-12-col-panel-pane-1">
          <div class="view view-promotion-slider-12-col view-id-promotion_slider_12_col view-display-id-panel_pane_1 view-dom-id-564713f4d738dd65125afa82f448335a">
            <div class="view-content">
              <div class="promo-slider-12-col"><div id="flexslider-1" class="flexslider flexslider-processed">
                <ul class="slides">
                  <li class="flex-active-slide" style="width: 100%; float: left; margin-right: -100%; position: relative; opacity: 1; display: block; z-index: 2;" aria-hidden="false">
                    <div class="promo-slider-12-col-item">
                      <div class="promo-slider-12-col-img">
                        <img typeof="foaf:Image" src="<?php print $front_page . drupal_get_path('theme', $themename); ?>/images/utlc-hero-02.jpg" alt="hero image of UTLC activities" title="" draggable="false" width="1080">
                      </div>
                      <div class="promo-slider-12-col-info">
                        <div class="promo-slider-12-col-inner">
                           <p style="margin-top: 1em;">The University Teaching & Learning Center (UTLC) is a faculty-driven effort to expand course design and teaching support to all GW faculty members.</p>
                           <span class="heroHeroBtn">
                              <a href="/utlc/about" target="_self">Read more</a>.
                           </span>
                        </div>
                      </div>
                    </div>
                  </li>
                </ul>
                <ol class="flex-control-nav flex-control-paging"></ol>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    </div>
<!--
    <div id="hero-image">
      <img alt="hero image of UTLC activities" src="<?php print $front_page . drupal_get_path('theme', $themename); ?>/images/utlc-hero-02.jpg"/>
    </div>
-->
  <?php endif; ?>

  <?php
    // Sets placeholder text for "all-search (bento)". Calls function from catalog_pointer module.
    if (function_exists('search_placeholder')) {
      if (search_placeholder() != '') {
        $bentoPlaceholder = search_placeholder();
      }
      else {
        $bentoPlaceholder = 'feminism, Les Miserables, economics, JSTOR, 3-D printing ...';
      }
    }
    else {
      $bentoPlaceholder = 'feminism, Les Miserables, economics, JSTOR, 3-D printing ...';
    }
  ?>

<!-- Internal Page Searchbox -->
<?php
  if((!$variables['is_front']) && ($campus !== 'utlc')) {
    include 'lai-header-in-search.php';
  }
?>

<!-- Homepage Searchbox -->
<?php if(drupal_is_front_page()): ?>
  <div id="home-search">
    <h3>What would you like to find at the libraries today?</h3>
    <form id="search-form" onsubmit="return false;">
    <div class="search-dropdown">
      <div class="search-dropdown-inner">
      <a href="#" aria-haspopup="true" class="current-search-text">Search All</a>
      <ul class="search-dropdown-ul">
      <li id="search-all" data-placeholder="<?php echo $bentoPlaceholder; ?>">
            <span class="search-label" tabindex="0">Search All</span>
            <div class="search-description">Articles and books, plus library databases, <a href="http://libguides.gwu.edu/" tabindex="-1">research guides</a> and tutorials</div>
        </li>
        <li id="search-articlesplus" data-placeholder="Fidel Castro, sustainable energy, gender and identity ...">
            <span class="search-label" tabindex="0">ArticlesPlus</span>
          <div class="search-description">Journal &amp; newspaper articles, plus books and more</div>
        </li>
        <li id="search-catalog" data-placeholder="The Communist Manifesto, calculus, Blade Runner ...">
            <span class="search-label" tabindex="0">Books & More</span>
        <div class="search-description">Books (including e-books), A/V media, and archival resources <div class="search-subdescription">Search the <a href="http://catalog.wrlc.org/">classic catalog</a></div></div>
      </li>
      <li id="search-journals" data-placeholder="Wall Street Journal, Journal of American History, sociology, ...">
        <span class="search-label" tabindex="0">Browse Journals</span>
        <div class="search-description">Online access to journals and other periodicals, by subject area and title</div>
      </li>
      <li id="search-website" data-placeholder="building hours, study rooms, Churchill ...">
        <span class="search-label" tabindex="0">Library Website</span>
        <div class="search-description">Library policies, news and events, and research help</div>
        </li>
      </ul>
      </div>
    </div>

    <input type="text" id="searchtext" aria-label="searchbox: enter your search terms here" placeholder="<?php echo $bentoPlaceholder; ?>"/>
    <select id="catalog-options">
      <option>title</option>
      <option>journal title</option>
      <option selected="selected">keyword</option>
      <option>author</option>
      <option>subject</option>
      <option>call number</option>
    </select>
    <select id="journals-options">
      <option>title begins with</option>
      <option>title (exact)</option>
      <option selected="selected">title keywords</option>
      <option>ISSN</option>
    </select>
      <input type="submit" value="Go" id="submitButton"/>
    </form>
    <p id="home-search-explanation">
      Articles and books, plus library databases, <a href="http://libguides.gwu.edu/">research guides</a> and tutorials
    </p>

  </div>

<?php endif; ?>

</div> <!-- closing hero div -->

<?php if(!drupal_is_front_page() && $campus !== 'utlc'): ?>
		<div class="promotional-six-col" id="internal-featured-services">
				<div class="promo-items-item">
                                <div class="promo-label">
				<div class="promo-image"><a href="/help/reference/ask-us" onclick="ga('send', 'event', 'service links', 'ask-us (internal <?php echo $campus; ?>)');"><img alt="icon: ask us for help" src="<?php print $front_page . drupal_get_path('theme', $themename); ?>/images/icons/icon-ask-us.png"></a></div>
								<h4 class="promo-title"><a href="/help/reference/ask-us" onclick="ga('send', 'event', 'service links', 'ask-us (internal <?php echo $campus; ?>)');">Ask Us</a></h4>
</span>
							</div>
                                </div>
				<div class="promo-items-item">
                                <div class="promo-label">
				<div class="promo-image"><a href="/about/facilities/study-room-reservations" onclick="ga('send', 'event', 'service links', 'rooms (internal <?php echo $campus; ?>)');"><img alt="icon: reserve a study room" src="<?php print $front_page . drupal_get_path('theme', $themename); ?>/images/icons/icon-study-rooms.png"></a></div>
								<h4 class="promo-title"><a href="/about/facilities/study-room-reservations" onclick="ga('send', 'event', 'service links', 'rooms (internal <?php echo $campus; ?>)');">Study Room Reservations</a></h4>
</span>
							</div>
                                </div>
				<div class="promo-items-item">
                                <div class="promo-label">
				<div class="promo-image"><a href="https://mylibrary.wrlc.org/" onclick="ga('send', 'event', 'service links', 'my account (internal <?php echo $campus; ?>)');"><img alt="icon: my library account" src="<?php print $front_page . drupal_get_path('theme', $themename); ?>/images/icons/icon-my-account.png"></a></div>
									<h4 class="promo-title"><a href="https://mylibrary.wrlc.org/" onclick="ga('send', 'event', 'service links', 'my account (internal <?php echo $campus; ?>)');">My Account</a></h4>
</span>
							</div>
                                </div>
				<div class="promo-items-item">
                                <div class="promo-label">
				<div class="promo-image"><a href="http://libguides.gwu.edu/databases" onclick="ga('send', 'event', 'service links', 'databases (internal <?php echo $campus; ?>)');"><img alt="icon: access our databases" src="<?php print $front_page . drupal_get_path('theme', $themename); ?>/images/icons/icon-databases.png"></a></div>
									<h4 class="promo-title"><a href="http://libguides.gwu.edu/databases" onclick="ga('send', 'event', 'service links', 'databases (internal <?php echo $campus; ?>)');">Databases</a></h4>
</span>
							</div>
                                </div>
				<div class="promo-items-item">
                                <div class="promo-label">
				  <div class="promo-image"><a href="/news-events/upcoming-events?field_event_format_value=1" onclick="ga('send', 'event', 'service links', 'workshops (internal <?php echo $campus; ?>)');"><img alt="icon: workshops" src="<?php print $front_page . drupal_get_path('theme', $themename); ?>/images/icons/icon-workshops.png"></a></div>
									<h4 class="promo-title"><a href="/news-events/upcoming-events?field_event_format_value=1" onclick="ga('send', 'event', 'service links', 'workshops (internal <?php echo $campus; ?>)');">Workshops</a></h4>
</span>
				  </div>
                                </div>
				<div class="promo-items-item has-promo-body">
                                <div class="promo-label">
				<div class="promo-image"><a href="#"><img alt="icon: all GW libraries" src="<?php print $front_page . drupal_get_path('theme', $themename); ?>/images/icons/icon-all-libraries.png"></a></div>
									<h4 class="promo-title"><a href="#">All GW Libraries</a></h4>

							</div>
<span class="promo-body">
	<ul>
	<li><a href="/">Estelle and Melvin Gelman Library</a></li>
	<li><a href="/eckles">Eckles Library at the Mount Vernon Campus</a></li>
	<li><a href="/virginia">Virginia Science and Technology Campus Library</a></li>
	</ul>
	<hr/>
	<ul>
	<li><a target="_blank" href="https://himmelfarb.gwu.edu/" class="external-link">Himmelfarb Health Sciences Library <i class="fa fa-external-link" aria-hidden="true"></i></a></li>
	<li><a target="_blank" href="https://www.law.gwu.edu/library" class="external-link">Jacob Burns Law Library <i class="fa fa-external-link" aria-hidden="true"></i></a></li>
	</li>
	</ul>
</span>

						</div>
                                            </div>


<?php endif; ?>

<script type="text/javascript">

jQuery(".search-dropdown").on("click", ".search-dropdown-ul li", function() {
  searchDropdown(jQuery(this));
});
jQuery(".search-label").on("keypress", function(e) {
  if (e.which == 13) { // Enter
    searchDropdown(jQuery(this).parent());
    e.preventDefault();
  }
});

function searchDropdown(passedThis) {
  jQuery(".current-search-text").text(passedThis.find(".search-label").text());
  jQuery("#search-form input[type=text]").attr("placeholder",passedThis.data("placeholder"));
  jQuery("#home-search-explanation").html(passedThis.find(".search-description").html());
  jQuery(".search-dropdown ul").hide();
  jQuery("#home-search-explanation").show();
  jQuery("#search-form input[type=text]").focus();
  jQuery("#catalog-options").hide();
  jQuery("#journals-options").hide();
  if (passedThis.attr("id") == "search-catalog") {
    jQuery("#catalog-options").show();
  }
  if (passedThis.attr("id") == "search-journals") {
    jQuery("#journals-options").show();
  }
}

jQuery(".search-dropdown").on("click", "a", function(e) {
  e.preventDefault();
});

jQuery(".search-dropdown").on("keydown", function(e) {
  var ul = jQuery(this).find("ul");
  if (ul.is(":hidden")) {
    ul.show();
  }
  var searchLabels = jQuery(".search-label");
  var nearestLabel = searchLabels.filter(":focus");
  var nearestLabelIndex = searchLabels.index(searchLabels.filter(":focus"));
  if (e.which == 38) { // Up arrow
    if (nearestLabel.length == 0) {
      searchLabels.last().focus();
    } else {
      searchLabels.eq(nearestLabelIndex - 1).focus();
    }
    e.preventDefault(); // page scrolling
  }
  if (e.which == 40) { // Down arrow
    if (nearestLabelIndex == searchLabels.length - 1) {
      searchLabels.first().focus();
    } else {
      searchLabels.eq(nearestLabelIndex + 1).focus();
    }
    e.preventDefault(); // page scrolling
  }
});

jQuery(".search-dropdown").on("mouseenter mouseover", function() {
  jQuery(this).find("ul").show();
});
jQuery(".search-dropdown").on("mouseleave", function() {
  jQuery(this).find("ul").hide();
});

</script>
